test/cppsuite/tests/search_near_01.cxx

```diff
@@ -170,116 +173,130 @@ class search_near_01 : public test_harness::test {
-                runtime_monitor::get_stat(
-                  tc->stat_cursor, WT_STAT_CONN_CURSOR_SEARCH_NEAR_PREFIX_FAST_PATHS, &prefix_stat);
-                logger::log_msg(LOG_INFO,
-                  "Read thread {" + std::to_string(tc->id) +
-                    "} skipped entries: " + std::to_string(entries_stat - prev_entries_stat) +
-                    " prefix fash path:  " + std::to_string(prefix_stat - prev_prefix_stat));
-
-                /*
-                 * Due to the concurrency of multiple threads and how WiredTiger increments the
-                 * entries skipped stat, it is possible that a thread can perform multiple search
-                 * nears before another can finish one. Account for this problem by creating a
-                 * buffer taking the maximum of either calculated 2 * expected entries or the
-                 * minimum expected entries. The minimum expected entries is necessary in the case
-                 * that expected entries is a low number.
-                 *
-                 * Assert that the number of expected entries is the maximum allowed limit that the
-                 * prefix search nears can traverse.
-                 */
-                buffer = std::max(2 * expected_entries, MINIMUM_EXPECTED_ENTRIES);
-                testutil_assert((expected_entries + buffer) >= entries_stat - prev_entries_stat);
-
-                /*
-                 * There is an edge case where we may not early exit the prefix search near call
-                 * because the specified prefix matches the rest of the entries in the tree.
-                 *
-                 * In this test, the keys in our database start with prefixes aaa -> zzz. If we
-                 * search with a prefix such as "z", we will not early exit the search near call
-                 * because the rest of the keys will also start with "z" and match the prefix. The
-                 * statistic will stay the same if we do not early exit search near.
-                 *
-                 * However, we still need to keep the assertion as >= rather than a strictly equals
-                 * as the test is multithreaded and other threads may increment the statistic if
-                 * they are searching with a different prefix that will early exit.
-                 */
-                if (srch_key == "z" || srch_key == "zz" || srch_key == "zzz") {
-                    testutil_assert(prefix_stat >= prev_prefix_stat);
-                } else {
-                    testutil_assert(prefix_stat > prev_prefix_stat);
-                }
+            tm.join();
 
-                tc->transaction.add_op();
-                tc->sleep();
+            /* Cleanup our workers. */
+            for (auto &it : workers) {
+                delete it;
+                it = nullptr;
             }
-            /* Reset our cursor to avoid pinning content. */
-            testutil_check(cursor->reset(cursor.get()));
+
```

Comment:
Do you need to clear `workers` vector after deleting and nullifying every entry in it?

Reply:
Yes we do! Thanks for catching that